worker/storageprovisioner/filesystem_ops.go

```diff
@@ -209,50 +218,66 @@ func destroyFilesystems(ctx *context, ops map[names.FilesystemTag]*destroyFilesy
 	for sourceName, filesystemParams := range paramsBySource {
 		logger.Debugf("destroying filesystems from %q: %v", sourceName, filesystemParams)
 		filesystemSource := filesystemSources[sourceName]
-		validFilesystemParams, validationErrors := validateFilesystemParams(filesystemSource, filesystemParams)
-		for i, err := range validationErrors {
-			if err == nil {
-				continue
-			}
-			statuses = append(statuses, params.EntityStatusArgs{
-				Tag:    filesystemParams[i].Tag.String(),
-				Status: status.Error.String(),
-				Info:   err.Error(),
-			})
-			logger.Debugf(
-				"failed to validate parameters for %s: %v",
-				names.ReadableString(filesystemParams[i].Tag), err,
-			)
+		destroyFilesystemParams := make([]params.DestroyFilesystemParams, len(filesystemParams))
+		for i, args := range filesystemParams {
+			destroyFilesystemParams[i] = destroyFilesystemParamsByTag[args.Tag]
 		}
-		filesystemParams = validFilesystemParams
-		if len(filesystemParams) == 0 {
+		if len(destroyFilesystemParams) == 0 {
 			continue
 		}
-		filesystemIds := make([]string, len(filesystemParams))
-		for i, filesystemParams := range filesystemParams {
-			filesystem, ok := ctx.filesystems[filesystemParams.Tag]
-			if !ok {
-				return errors.NotFoundf("filesystem %s", filesystemParams.Tag.Id())
+		destroyFilesystemTags := make([]names.FilesystemTag, 0, len(destroyFilesystemParams))
```

Comment:
This method's become really long and is shuffling things around in several different data structures. Maybe split this part (partitioning into release and destroy groups, and then executing the groups) into a separate method?